src/script/ismine.h

```diff
@@ -9,22 +9,44 @@
 #include <script/standard.h>
 
 #include <stdint.h>
+#include <bitset>
 
 class CKeyStore;
 class CScript;
 
 /** IsMine() return codes */
 enum isminetype
 {
-    ISMINE_NO = 0,
-    ISMINE_WATCH_ONLY = 1,
-    ISMINE_SPENDABLE = 2,
-    ISMINE_ALL = ISMINE_WATCH_ONLY | ISMINE_SPENDABLE
+    ISMINE_NO         = 0,
+    ISMINE_WATCH_ONLY = 1 << 0,
+    ISMINE_SPENDABLE  = 1 << 1,
+    ISMINE_ALL        = ISMINE_WATCH_ONLY | ISMINE_SPENDABLE,
+    ISMINE_ENUM_ELEMENTS,
 };
 /** used for bitflags of isminetype */
 typedef uint8_t isminefilter;
 
 isminetype IsMine(const CKeyStore& keystore, const CScript& scriptPubKey);
 isminetype IsMine(const CKeyStore& keystore, const CTxDestination& dest);
 
+struct CachableAccount
+{
+    enum class Type : uint8_t {
+        CREDIT,
+        DEBIT,
+    } const m_type;
+    std::bitset<ISMINE_ENUM_ELEMENTS> m_cached;
+    CAmount m_value[ISMINE_ENUM_ELEMENTS];
+    explicit CachableAccount(Type type) : m_type(type) {}
+    void Reset()
+    {
+        m_cached.reset();
```

Comment:
Since `CachableAccount` members are public you could ditch `Reset` and call `m_cached.reset()` directly. Otherwise you could make members private and add `IsCached()`,  `Get()` and `Type()` accessors.

Reply:
I think leaving it public is ok, but I can be convinced to make it private. For now, I made `Reset` inline, which basically has the same effect as what you are suggesting without exposing too many innards.